### PR TITLE
Docs x12 connector technical edit mc

### PR DESCRIPTION
**Pull request by mcherem**

### CODE REVIEW

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
-|Transactions |A hierarchy of the transaction sets that are read by the module during a read operation, or are to be sent by the module during a write operation. The value of this top-level key is a map with standard versions as keys, in the form 'v005010`.
-For example, if you are using version 005010 850 and 855 transaction sets, the Transactions would contain a map with one key, `v005010`. The value of this key would be another map, one with two keys, "850" and "855". Each of these would contain a list of individual 850 and 855 transaction sets that can be processed. If you're using 997 functional acknowledgments `997` is the key for a list of 997 transaction sets.
+|Delimiters |The delimiters used for the message. If used, this set is based on the last interchange processed by a read operation. If set for a write operation, this overrides the values in the module configuration. The characters in the string are interpreted based on position in the following order: data separator, sub-element separator, repetition separator, or `U` if none, segment terminator.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc:74]: if one what?

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
+|Errors (read only) |A list of errors that are associated with the input as a whole, or with interchanges with no valid transaction sets
+|FunctionalAcksGenerated (read only) |A list of `997` or `999` configured acknowledgments that are generated by the module during the read operation
+|Group (write only) |Map of GS group header segment values used as defaults when writing groups
+|Interchange (write only) |Map of ISA interchange header segment values used as defaults when writing interchanges
+|InterchangeAcksGenerated (read only) |A list of TA1 acknowledgments that were generated by the module during the read operation
+|InterchangeAcksReceived (read only) |A list of TA1 acknowledgments that were received by the module during the read operation
+|InterchangeAcksToSend (write only) |A list of TA1 acknowledgments that are to be sent by the module during the write operation
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: "are" sent or "are to be sent"?

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
+|Transactions |A hierarchy of the transaction sets that are read by the module during a read operation, or are sent by the module during a write operation. The value of this top-level key is a map with standard versions as keys, in the form of `v005010`.
+For example, if you are using version `005010` `850` and `855` transaction sets, the Transactions would contain a map with one key, `v005010`. The value of this key would be another map, one with two keys, `850` and `855`. Each of these  contain a list of individual `850` and `855` transaction sets that are processed. If you're using `997` functional acknowledgments. then `997` is the key for a list of 997 transaction sets
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: Missing a period in the last sentence

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
-|Name |Transaction set name.
+|Detail |Map of segments or loops from the detail section of the transaction set. Values are maps for segments or loops that occur once, lists of maps for values that are repeated.
+|Errors (read only) |A list of errors that are associated with the transaction set
+|Group |Map of GS group header segment values. When reading a message, this map is the actual data from the enclosing group, that is, a single map linked from all transaction sets in the group. When writing a message, these values are used for creating the enclosing group. Values not populated in this map then default to the values from the Group map at the message level
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: Missing period

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
+|Heading |Map of segments or loops from the heading section of the transaction set. Values are maps for segments or loops that occur once. Lists of maps are for values that are repeated.
+|Id |Transaction ID, this must match the key of the containing transaction list
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: this > which

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
+|Interchange |Map of ISA interchange header segment values. For functional acknowledgments generated by receive processing, this map is a copy of the data for the containing interchange with sender and receiver identification components (ISA05/ISA06 and ISA07/ISA08) interchanged. When writing a message, these values are used for creating the enclosing interchange. That is, gathering transactions with the same interchange values into a single interchange, regardless of whether the actual maps are the same. Values not populated in this map default to the values from the Interchange map at the message level.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: The sentence starting with "That is" is not a sentence.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc [hunk 7]
```diff
@@ -26,124 +26,127 @@ To use the X12 EDI connector in a production environment, you must have:
...
 
-////
-List and link to up to 10 Exchange templates and examples.
-Use the Integration Pattern categories (broadcast, migration, bidirectional sync, aggregation).
-////
+=== X12 Connector Templates in Exchange
+
+https://www.mulesoft.com/exchange/org.mule.templates/template-b2b-edi-outbound-x12-map/[Map application messages to outbound EDI X12 transactions]
+
+Use this for setting up your Outbound transactions.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/index.adoc]: Can you say "Sets up your Outbound transactions" instead?

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 8]
```diff
@@ -1,23 +1,26 @@
...
+* X12 HIPAA EDI message reading, message validation, and message writing
 
-* X12 HIPAA EDI message-reading, message-validation, and message-writing
 * Integration with DataSense and DataWeave
-* HIPAA message packs for 005010X212, 005010X214B3, 005010X217, 005010X218, 005010X220A1, 005010X221A1, 005010X222A1, 005010X222A2, 005010X223A2, 005010X223A3, 005010X224A2, 005010X224A3, and 005010X279A1
+
 * The ability to customize the base X12 HIPAA schemas
 
-Most aspects of using HIPAA with the X12 Module are the same as using standard X12. The main differences are:
+Most aspects of using HIPAA with the X12 Connector are the same as using standard X12. The main differences are:
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: "with X12 Connector" (Kim always has us remove "the."

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 9]
```diff
@@ -30,29 +33,33 @@ HIPAA schemas also include details of code sets defined for elements. Data is va
...
 
-The X12 Module uses a YAML format called ESL (for EDI Schema Language) to represent both standard X12 and HIPAA schemas.
+The X12 Module uses a YAML format called EDI Schema Language (ESL) to represent both standard X12 and HIPAA schemas.
 The HIPAA standards are based on X12, but modify the corresponding X12 base standards in several aspects including:
 
-* Multiple specialized definitions of HL (Hierarchical Level) loops, where a basic loop structure defined in X12 is expanded into several variations, often nested.
-* Modular loop definitions that can be reused at different points in the definition.
-* Distinguishing between different uses of a repeated segment by code values in a particular element.
-* Changing usage requirements for segments and segment components.
+* Multiple specialized definitions of Hierarchical Level (HL) loops, where a basic loop structure defined in X12 is expanded into several variations. These are often nested
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: This sentence is missing something. It is not clear.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 10]
```diff
@@ -30,29 +33,33 @@ HIPAA schemas also include details of code sets defined for elements. Data is va
...
+
+* Modular loop definitions that can be reused at different points in the definition
+
+* Distinguishing between different uses of a repeated segment by code values in a particular element
+
+* Changing usage requirements for segments and segment components
 
 These features of the HIPAA ESL schema definitions are very different from standard X12 schemas.
-This partial example illustrates this. It is from the schema for the 005010X222A2 837 transaction set:
+This partial example illustrates this. It is from the schema for the `005010X222A2` `837` transaction set:
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:63]: illustrates what?

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 11]
```diff
@@ -116,19 +123,19 @@ structures:
 
 == About Areas Key
 
-In the previous schema example, the areas key has a value array of individual area definitions. Areas are similar to the breakdown of normal X12 transaction sets into header, detail, and summary sections, but provide much finer granularity.  Instead of the three fixed portions of a transaction set defined in X12, there may be twenty or more areas defined in a HIPAA transaction set.
+In the previous schema example, the areas key has a value array of individual area definitions. Areas are similar to the breakdown of normal X12 transaction sets, for example into header, detail, and summary sections, but provide much finer granularity.  Instead of the three fixed portions of a transaction set as defined in X12 there may be twenty or more areas defined in a HIPAA transaction set.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: I would put commas around "as defined in X12."

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 12]
```diff
@@ -116,19 +123,19 @@ structures:
...
 
 Each area is a reusable component of the definition, and is identified by a code character value, which by convention can be a single digit or single alpha character.
 
-Areas are referenced for inclusion in the definition with an area item in the component list. In an X12 schema definition, the list of components of a group or area may contain only segments, groups, and a group variation called wrapped (used for LS/LE loops, in X12 terms). In a HIPAA schema definition, the list of components may also contain area references. The effect of referencing an area is the same as if all the components of the area were inserted in the definition at the point of the reference.
+Areas are referenced for inclusion in the definition with an area item in the component list. In an X12 schema definition, the list of components of a group or area can contain only segments, groups, and a group variation called wrapped (used for LS/LE loops, in X12 terms). In a HIPAA schema definition, the list of components may also contain area references. The effect of referencing an area is the same as if all the components of the area were inserted in the definition at the point of the reference.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: may > can

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 13]
```diff
@@ -146,11 +153,11 @@ The following is another portion of the same 005010X222A2 837 transaction set sc
...
 
-The X12 EDI Connector enforces these code sets for HIPAA documents, signaling an error if a transaction set uses an undefined value for a field (that is, a value not listed as a key in the `codeSet`) either when parsing or writing. In some cases, such as the first element of the BHT definition, only a single value is allowed. In other cases there may be many potential values.
+The X12 EDI Connector enforces these code sets for HIPAA documents, signaling an error if a transaction set uses an undefined value for a field (that is, a value not listed as a key in the `codeSet`) either when parsing or writing. In some cases, such as the first element of the `BHT` definition, only a single value is allowed. In other cases there can be many potential values.
 
 == About Segment Variants
 
-The following is a third portion of the same 005010X222A2 837 transaction set schema example, this one showing two different DTP segment definitions:
+The following is a third portion of the same `005010X222A2` `837` transaction set schema example, this one showing two different DTP segment definitions:
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Run-on sentence

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 14]
```diff
@@ -170,15 +177,15 @@ The following is a third portion of the same 005010X222A2 837 transaction set sc
   - { id: '1251', name: 'Acute Manifestation Date', usage: M, type: AN, minLength: 1, maxLength: 35 }
 ----
 
-These two definitions apply to different instances of the DTP segment, as part of the 2300 Claim Information loop. In the transaction set structure these uses of the DTP segment occur essentially in the same position, matching two possible occurrences of a repeating DTP segment in the base X12 standard. But because the two uses of the segment are supplying different information the HIPAA standard gives them different names and interprets the `DTP03` field in different ways.
+These two definitions apply to different instances of the `DTP` segment, as part of the 2300 Claim Information loop. In the transaction set structure these uses of the `DTP` segment occur essentially in the same position, matching two possible occurrences of a repeating DTP segment in the base X12 standard. Because the two uses of the segment are supplying different information the HIPAA standard gives them different names and interprets the `DTP03` field in different ways.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Need a comma after "are supplying different information"

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 15]
```diff
@@ -170,15 +177,15 @@ The following is a third portion of the same 005010X222A2 837 transaction set sc
...
 
 In this case the data value in the first field of the segment, the `Date Time Qualifier` field, identifies which variation of the segment is actually being used. Since the code set for this field has different values for each of these uses the value present in the field tells whether the DTP segment in a parsed document is a `DTP_DateAccident` or a `DTP_DateAcuteManifestation` (or any of several other uses of the DTP segment in the same position). The `varValue: true` flag in the schema definition indicates that this first field is used in this way to distinguish between variations.
 
-Note that even though the value of this field is effectively fixed for each use of the segment, it still must be set correctly when writing data. If you supply a different value for this field, or don't supply a value, you'll get an error when writing.
+Even though the value of this field is effectively fixed for each use of the segment, it needs to be set correctly when writing data. If you supply a different value for this field, or don't supply a value, you'll get an error when writing.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: "needs to" > "must"

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 16]
```diff
@@ -206,27 +213,24 @@ specifications, and the list of items gives the numbers of the values governed b
...
 == Modifying Schemas
 
 Due to the differences between standard X12 and HIPAA schemas the use of overlay schemas to modify a base
-definition is not supported for HIPAA. Instead, the recommended approach to use for modifications is to
-extract the supplied HIPAA schema from inside the `x12-schemas-2.0.0.jar`, that is found in the standard MuleSoft enterprise Maven repositories (under group ID com.mulesoft.connectors).
-You can copy a message structure schema from this JAR (that contains both standard X12 and HIPAA schemas) and modify the extracted schema to use
-directly. Unlike X12 schemas, which as supplied use a base set of segment, composite, and element
-definitions, the HIPAA schemas are self-contained. This makes it easy to make changes to the schema
-without needing to work with multiple files.
+definition is not supported for HIPAA. Instead, for modifications extract the supplied HIPAA schema from inside the `x12-schemas-2.0.0.jar`. This is found in the standard MuleSoft enterprise Maven repositories under group ID com.mulesoft.connectors.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Need a comma after "for modifications"

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 17]
```diff
@@ -206,27 +213,24 @@ specifications, and the list of items gives the numbers of the values governed b
...
+You can copy a message structure schema from this JAR (it contains both standard X12 and HIPAA schemas) and modify the extracted schema to use it directly. Unlike X12 schemas, that use a base set of segment, composite, and element definitions, the HIPAA schemas are self-contained. This makes it easy to make changes to the schema without needing to work with multiple files.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: No comma before "that"

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 18]
```diff
@@ -206,27 +213,24 @@ specifications, and the list of items gives the numbers of the values governed b
...
 This validates the basic syntactical integrity of the EDI submission.
 * Type 2: HIPAA syntactical requirement testing – Testing for HIPAA Implementation Guide-specific syntax requirements, such as limits on repeat counts, used and not used qualifiers, codes, elements, and segments. Also included in this type is testing for HIPAA required or intra-segment situational data elements, testing for non-medical code sets as laid out in the WEDI SNIP implementation guide, and values and codes noted in the WEDI SNIP implementation guide using an X12 code list or table.
-** As the connector cannot determine a course of action for intra-segment situational data elements, intra-segment situational data elements aren't covered by the X12 EDI connector. Users need to set a validation logic outside of the connector.
+** As the connector cannot determine a course of action for intra-segment situational data elements, intra-segment situational data elements aren't covered by the X12 connector. Users need to set a validation logic outside of the connector.
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: This should not be a bullet. It should be a bullet continuation.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-config-topics.adoc [hunk 19]
```diff
@@ -34,23 +32,22 @@ To configure schemas in Anypoint Studio:
...
-|stringCharacterSet="BASIC"/"EXTENDED"/"UNRESTRICTED" |Use a substitution character replacement for invalid characters, or reject them as errors if none is set or enabled for receive messages in the parser options.  The default is `EXTENDED` for string data.
-|stringSubstitutionChar="?" |Substitution character used to replace invalid characters in string values. The default is `none`.
-|versionIdentifierSuffix="XYZ" |Version identifier code suffix for the group (GS). Setting this value affects both send and receive message handling. On the receive side, the GS08 version code must contain the X12 schema version (for example, 004010, 005010, and so forth) followed by this suffix. Not setting this value means the receive-only just verifies that the GS version identifier code starts with the configured schema version. On the send side, the combination of the X12 schema version and this suffix comprises the GS08 value.
+|formValidation="X12"/"HIPAA_SNIP1"/"HIPAA_SNIP2" |The schema form and validation level can be either X12 or HIPAA. The default is X12
+|schemas=list of values |List of schema paths the connector can use. These paths can be either a file system or `classpath`
```
Comment (dhirsch1) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-config-topics.adoc]: You are missing a period in the sentence here and elsewhere.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 2 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 3 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 4 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 5 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 6 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/index.adoc (hunk 7 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/index.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 8 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed throughout.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 10 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:63]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 11 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 12 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 14 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 15 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 16 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 17 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 18 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-config-topics.adoc (hunk 19 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-config-topics.adoc]: Fixed.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 9 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: REworded.

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc (hunk 13 above)
Comment (mcherem) [reply x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-hipaa.adoc]: rewritten.

**Review by dhirsch1 — DISMISSED** [review]

**Review by kevintroller — APPROVED** [review]

**Pull request merged**